### PR TITLE
Introduce a new HTTP endpoint: /accounts/{pubkey}/next-nonce

### LINKED ISSUE #3662: HTTP endpoint: get next nonce
Sending transactions for the same account from a distributed system requires knowledge for the latest account `nonce`. Since transactions can be included, rejected or even replaced by transactions with a competing nonce, the only source of truth shall be either a wallet or a synced node.

Provide an endpoint that returns what the next nonce for an account would be, taken into consideration transactions in the pool, originating from the same account.

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #3662

The work on this PR is  being suppoeted by ACF.

### CODE REVIEW

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 1]
```diff
@@ -3015,15 +3029,20 @@ get_transaction(_Config) ->
     {ok, SpendTxBin} = aeser_api_encoder:safe_decode(transaction, EncodedSpendTx),
     SpendTx = aetx:deserialize_from_binary(SpendTxBin),
     {ok, SignedSpendTx} = aecore_suite_utils:sign_on_node(?NODE, SpendTx),
+    Nonce = aetx:nonce(aetx_sign:tx(SignedSpendTx)),
+    {Nonce, Nonce} = {Nonce, NextNonce},
```
Comment (earlyriser99) [inline apps/aehttp/test/aehttp_integration_SUITE.erl:3033]: Just checking - this is simply an assert right?

File: apps/aehttp/priv/oas3.yaml [hunk 2]
```diff
@@ -543,6 +543,49 @@ paths:
...
             application/json:
               schema:
+                $ref: "#/components/schemas/Error"
+  "/accounts/{pubkey}/next-nonce":
+    get:
+      tags:
+        - external
+        - account
+      operationId: GetAcountNextNonce
+      description: Get an account's next nonce; This is being computed according to whatever is the current account nonce and what transactions are currently present in the transaction pool
```
Comment (uwiger) [inline apps/aehttp/priv/oas3.yaml:553]:
```suggestion
      description: Get an account's next nonce; This is computed according to whatever is the current account nonce and what transactions are currently present in the transaction pool
```

File: apps/aehttp/priv/oas3.yaml [hunk 3]
```diff
@@ -543,6 +543,49 @@ paths:
...
+      parameters:
+        - $ref: '#/components/parameters/intAsString'
+        - $ref: '#/components/parameters/accountPubkey'
+        - in: query
+          name: "strategy"
+          description: "What strategy to use in order to determine the next nonce: shall it check for continuity or return the largest nonce seen + 1. If you choose strategy max - the greatest nonce seen in the account or currently in the transaction pool is incremented with 1 and returned. If you choose the strategy continuity - transactions in the mempool are checked if there are gaps - missing nonces that prevent transactions with greater nonces to get included."
```
Comment (uwiger) [inline apps/aehttp/priv/oas3.yaml:559]:
```suggestion
          description: "What strategy to use in order to determine the next nonce: shall it check for continuity or return the largest nonce seen + 1. If you choose strategy max, then the greatest nonce seen in the account or currently in the transaction pool is incremented with 1 and returned. If you choose the strategy continuity, then transactions in the mempool are checked if there are gaps - missing nonces that prevent transactions with greater nonces to get included." 
```

File: apps/aehttp/priv/swagger.yaml [hunk 4]
```diff
@@ -568,6 +568,44 @@ paths:
           description: 'Account not found'
           schema:
             $ref: '#/definitions/Error'
+  /accounts/{pubkey}/next-nonce:
+    get:
+      tags:
+        - external
+        - account
+      operationId: GetAcountNextNonce
+      description: "Get an account's next nonce; This is being computed according to whatever is the current account nonce and what transactions are currently present in the transaction pool"
```
Comment (uwiger) [inline apps/aehttp/priv/swagger.yaml:577]:
```suggestion
      description: "Get an account's next nonce; This is computed according to whatever is the current account nonce and what transactions are currently present in the transaction pool"
```

File: apps/aehttp/priv/swagger.yaml [hunk 5]
```diff
@@ -568,6 +568,44 @@ paths:
...
+        - application/json
+      parameters:
+        - in: path
+          name: pubkey
+          description: 'The public key of the account'
+          required: true
+          type: string
+        - in: query
+          name: strategy
+          description: "What strategy to use in order to determine the next nonce: shall it check for continuity or return the largest nonce seen + 1. If you choose strategy max - the greatest nonce seen in the account or currently in the transaction pool is incremented with 1 and returned. If you choose the strategy continuity - transactions in the mempool are checked if there are gaps - missing nonces that prevent transactions with greater nonces to get included."
```
Comment (uwiger) [inline apps/aehttp/priv/swagger.yaml:588]:
```suggestion
          description: "What strategy to use in order to determine the next nonce: shall it check for continuity or return the largest nonce seen + 1. If you choose strategy max, then the greatest nonce seen in the account or currently in the transaction pool is incremented with 1 and returned. If you choose the strategy continuity, then transactions in the mempool are checked if there are gaps - missing nonces that prevent transactions with greater nonces to get included." 
```

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 6]
```diff
@@ -84,6 +84,7 @@ queue('GetGenerationByHeight')                  -> ?READ_Q;
 queue('GetAccountByPubkey')                     -> ?READ_Q;
 queue('GetAccountByPubkeyAndHeight')            -> ?READ_Q;
 queue('GetPendingAccountTransactionsByPubkey')  -> ?READ_Q;
+queue('GetAcountNextNonce')                     -> ?READ_Q;
```
Comment (uwiger) [inline apps/aehttp/src/aehttp_dispatch_ext.erl:87]:
```suggestion
queue('GetAccountNextNonce')                    -> ?READ_Q;
```

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 7]
```diff
@@ -422,6 +423,24 @@ handle_request_('GetPendingAccountTransactionsByPubkey', Params, _Context) ->
             {400, [], #{reason => <<"Invalid public key">>}}
     end;
 
+handle_request_('GetAcountNextNonce', Params, _Context) ->
```
Comment (uwiger) [inline apps/aehttp/src/aehttp_dispatch_ext.erl:426]:
```suggestion
handle_request_('GetAccountNextNonce', Params, _Context) ->
```

File: apps/aehttp/priv/swagger.yaml [hunk 8]
```diff
@@ -568,6 +568,44 @@ paths:
           description: 'Account not found'
           schema:
             $ref: '#/definitions/Error'
+  /accounts/{pubkey}/next-nonce:
+    get:
+      tags:
+        - external
+        - account
+      operationId: GetAcountNextNonce
```
Comment (uwiger) [inline apps/aehttp/priv/swagger.yaml:576]:
```suggestion
      operationId: GetAccountNextNonce
```

File: apps/aehttp/priv/oas3.yaml [hunk 9]
```diff
@@ -543,6 +543,49 @@ paths:
           content:
             application/json:
               schema:
+                $ref: "#/components/schemas/Error"
+  "/accounts/{pubkey}/next-nonce":
+    get:
+      tags:
+        - external
+        - account
+      operationId: GetAcountNextNonce
```
Comment (uwiger) [inline apps/aehttp/priv/oas3.yaml:552]:
```suggestion
      operationId: GetAccountNextNonce
```

**Review by uwiger — CHANGES_REQUESTED** [review]
Typo in operation id should be fixed

**Review by uwiger — DISMISSED** [review]

File: apps/aecore/src/aec_next_nonce.erl [hunk 10]
```diff
@@ -14,23 +15,62 @@
...
+                    {ok, NextNonce};
+                continuity ->
+                    {ok, AllTxs} = aec_tx_pool:peek(infinity, Pubkey),
+                    %% get all nonces from the pool sorted asc
+                    AllNoncesInPool = lists:sort([aetx:nonce(aetx_sign:tx(T)) ||
+                                                      T <- AllTxs]),
+                    %% prepend the account nonce this relies on it being
+                    %% smallest number (we don't keep invalid txs in the pool)
+                    AllNonces = [StateTreeNonce | AllNoncesInPool],
+                    Missing =
```
Comment (hanssv) [inline apps/aecore/src/aec_next_nonce.erl:39]: (Much) shorter: 
`Next = hd(lists:seq(StateTreeNonce, StateTreeNonce + length(AllNoncesInPool) + 1) -- AllNonces)`

**Review by uwiger — APPROVED** [review]

**Review by hanssv — APPROVED** [review]

**Pull request merged**